### PR TITLE
Add "deploy" subcommand

### PR DESCRIPTION
**Pull request by bernerdschaefer**
When a user runs `ENVIRONMENT deploy`

Parity will:

  - fetch the git remote for the environment
  - check if there are changes in `db/migrate`
    which will be pushed.
  - push local master branch to remote master
  - run parity's migrate command if needed.

### CODE REVIEW

**Comment by geoffharcourt** [comment]
This idea was considered a year ago ([great minds think alike!](https://github.com/croaky/parity/issues/9)), but the decision was to let parity do heroku and for git to do git.

For automigration, I'd recommend using Tim Pope's [Heroku Ruby Buildopack](https://github.com/tpope/heroku-buildpack-ruby-tpope), which runs the migration with the deploy for only a single restart and minimal downtime.

**Comment by bernerdschaefer** [comment]
Note that we've been adding `bin/deploy` scripts to new projects through [suspenders][]. One disadvantage to it living there is that not all projects have such a script. Also, defining `deploy` in parity lets us evolve the strategy (for example, unlike the deploy script from suspenders, this version won't restart if there are no migrations being deployed).

Finally, my brain has trouble swapping `bin/deploy production` with `production deploy`, which produces the following terrifying output:

```
> production deploy
 !    `deploy` is not a heroku command.
 !    Perhaps you meant `destroy`.
 !    See `heroku help` for a list of available commands.
```

[suspenders]: https://github.com/thoughtbot/suspenders/blob/master/templates/bin_deploy

**Comment by bernerdschaefer** [comment]
@croaky Would love to have you weigh in on this. I've been using and liking this over the last week. Faster deploys than `bin/deploy` when there are no pending migrations, and works on non-suspenders projects!

File: lib/parity/environment.rb [hunk 1]
```diff
@@ -100,5 +110,12 @@ def basename
     def pass_through
       [subcommand, arguments].join(' ').strip
     end
+
+    def skip_migration?
+      Kernel.system %{
+        git fetch #{environment} &&
+        git diff --quiet #{environment}/master..master -- db/migrate
```
Comment (croaky) [inline lib/parity/environment.rb]: Trying to think through whether there's any edge cases to consider with `production/master..master`. Anything terrible that can go wrong if I haven't fetched from `origin` or something? I think it's fine, just giving it some scrutiny.

**Comment by croaky** [comment]
I like that "migrate only when necessary" bit. @tute Something to consider for the default `bin/deploy` script in Suspenders.

I'm down with this idea. Sorry for originally shutting it down, @geoffharcourt. I guess I'm changing my tune with multiple people coming to the same conclusion.

@b Good to merge at your discretion.

**Pull request merged**